### PR TITLE
make creepslow modifier a class attribute

### PR DESCRIPTION
**Pull request by bmorel**
This includes default values matching the ones in 0.52.1 if nothing is found in configuration, to avoid breaking compat.

Note that alien classes are also checked, because I thought that it might be interesting to give aliens a speed boost when in their base. Can easily be removed and the option made humans-only.

### CODE REVIEW

**Comment by slipher** [comment]
What is the point of this?

**Comment by bmorel** [comment]
Several ones.

1. this fix a TODO in code
2. this makes things less hard-coded
3. with the removal of the 0.52 compat, lines are removed
4. with minor code changes, it would be possible to give aliens a speed bonus in their base, depending on their class
5. it makes it possible to have different maluses depending on the armour ratio of the human

**Comment by bmorel** [comment]
Guess it won't be merged.

**Comment by necessarily-equal** [comment]
Actually I think this needs a companion PR on unvanquished.dpkdir ?

**Comment by bmorel** [comment]
Technically it does not since it will adopt default values in absence of the right config keys (also, since I closed it, I will not maintain it, so you'll have to do the changes one might request)

File: src/shared/bg_parse.cpp [hunk 1]
```diff
@@ -1279,6 +1286,37 @@ void BG_ParseClassAttributeFile( const char *filename, classAttributes_t *ca )
...
+				break;
+			case PCL_HUMAN_NAKED:
+				ca->creepModifier = 0.5f;
+				break;
+			case PCL_HUMAN_LIGHT:
+			case PCL_HUMAN_MEDIUM:
+			case PCL_HUMAN_BSUIT:
+				ca->creepModifier = 0.75f;
+				break;
+			case PCL_NONE:
```
Comment (necessarily-equal) [inline src/shared/bg_parse.cpp:1315]:
```suggestion
```

File: src/shared/bg_parse.cpp [hunk 2]
```diff
@@ -1279,6 +1286,37 @@ void BG_ParseClassAttributeFile( const char *filename, classAttributes_t *ca )
 			            token, filename );
 		}
 	}
+
+	// TODO(0.53) move the values into config files
+	if ( !( defined & CREEPMODIFIER ) )
+	{
+		Log::Warn( "creepModifier not defined in %s (using previously hard-coded values)", filename );
+		switch( ca->number )
+		{
```
Comment (necessarily-equal) [inline src/shared/bg_parse.cpp:1295]:
```suggestion
		{
			case PCL_NONE:
```

**Review by necessarily-equal — COMMENTED** [review]
Can you just apply that?

**Review by necessarily-equal — APPROVED** [review]
approved for squash&merge

**Comment by necessarily-equal** [comment]
@slipher are you ok with merging this?

**Comment by slipher** [comment]
1. It's kind of perverse that the creep modifier attribute is required for alien configs, when it is impossible for them to have SS_CREEPSLOWED.
2. I don't want to have warning spam when using 0.52 assets. Why not put this on the 0.53 branch? Then you could do everything in 1 step without all the default values.
3. The companion PR should be opened before submitting this.

**Comment by bmorel** [comment]
> It's kind of perverse that the creep modifier attribute is required for alien configs, when it is impossible for them to have SS_CREEPSLOWED.

This can be fixed easily by making it a human-only thing. I'm not fond of that idea though, but you are right that the attribute SS_CREEPSLOWED should be applied to aliens if the configuration key works for them.
I think I'll just implement this, it's likely just replacing `ForEntities<HumanClassComponent>([&] (Entity& other, HumanClassComponent&) {` with... dunno what exactly, but I'll find out.

Before I try though, do you agree with that plan?
That is, to have aliens possibly affected by creep too (considering that it might be an actual speed increase, why not)?

> I don't want to have warning spam when using 0.52 assets. Why not put this on the 0.53 branch? Then you could do everything in 1 step without all the default values.

Fine with me.

> The companion PR should be opened before submitting this.

Ok.

**Comment by necessarily-equal** [comment]
> > It's kind of perverse that the creep modifier attribute is required for alien configs, when it is impossible for them to have SS_CREEPSLOWED.
>
> This can be fixed easily by making it a human-only thing. 

Or we can just call that a creepSpeedMultiplier, and apply it to everyone ?


Le 19 février 2022 14:34:51 GMT+01:00, "Morel Bérenger" ***@***.***> a écrit :

**Comment by bmorel** [comment]
That's what it would become in practice yes.

**Comment by slipher** [comment]
It wastes network bandwidth if you set a useless state flag that doesn't do anything.

**Comment by bmorel** [comment]
Well, I did closed that PR because I thought it would not be merged without endless discussions, too much for what the PR is worth.
The problem is, I regularly browse all PRs I have pending, just in case, and github does not give a shit that I'm not the person which opened this in last, so I'm more or less tricked in staying active in this.
@necessarily-equal I would really like it if you could clone than branch and do another PR which would supersede this one, so that this one can be closed and that I no longer see it in the PRs I'm the author of.

**Pull request closed without merging**